processing-service-core/api/models/tenant.js

```diff
@@ -0,0 +1,340 @@
+ *
+ * @param {String} tenantId String The ID of the tenant.
+ * no response value expected for this operation
+ **/
+function deleteTenantConfigs(tenantId) {
+    var tenantConfig = Q.defer();
+    tenantConfigStoreModel.getTenantConfigs(tenantId)
+            .then(function (tenantConfigs) {
+                var tenantConfigsMap = createMap(tenantConfigs);
+                for (var key in tenantConfigsMap) {
+                    tenantConfigStoreModel.deleteTenantConfig(tenantId, key);
+                }
+                logger.debug("Successfully deleted tenant specific config for tenant " + tenantId);
+                tenantConfig.resolve();
+            })
+            .all()
+            .fail(function (errorResponse) {
+                logger.debug("Unable to remove tenant specific config for tenant {}", tenantId);
+                tenantConfig.reject(errorResponse);
+            })
+            .done();
+    return tenantConfig.promise;
+}
+
+/**
+ * Retrieve effective tenant specific config.
+ * Returns the effective tenant specific config for this key. That can be either a custom value specified for the tenant or the default.
+ *
+ * @param {String} tenantId String The ID tenant.
+ * @param {String} key String The custom configuration key.
+ * returns effectiveTenantConfig
+ **/
+function getEffectiveTenantConfig(tenantId, key) {
+    var tenantConfig = Q.defer();
+    var effectiveConfig = {};
+    effectiveConfig.tenantId = tenantId;
+    effectiveConfig.key = key;
+    logger.debug("Retrieving effective tenant config for tenant " + tenantId + " with key " + key);
+    tenantConfigStoreModel.getTenantConfig(tenantId, key)
+            .then(function (tenantConfigResult) {
+                effectiveConfig.value = tenantConfigResult;
+                effectiveConfig.valueType = "CUSTOM";
+                tenantConfig.resolve(effectiveConfig);
+            })
+            .fail(function (errorResponse) {
+                logger.debug("No tenant specific config for tenant " + tenantId + " with key " + key + " was found, returning default value");
+                if (errorResponse instanceof ApiError && errorResponse.type === apiErrorTypes.ITEM_NOT_FOUND) {
+                    globalConfigStoreModel.getGlobalConfig(tenantId, key)
+                            .then(function (tenantDefaultConfig) {
+                                for (var key in tenantDefaultConfig) {
+                                    effectiveConfig.value = tenantDefaultConfig[key];
+                                    effectiveConfig.valueType = "DEFAULT";
+                                    tenantConfig.resolve(effectiveConfig);
+                                }
+                            })
+                            .fail(function (errorResponse) {
+                                tenantConfig.reject(errorResponse);
+                            })
+                            .done();
+                } else {
```

Comment:
To reduce the nesting here can you invert this condition so that it rejects and returns if the error response isn't an ApiError or type ITEM_NOT_FOUND?